src/System.Text.Primitives/System/Text/InvariantParser_float.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class InvariantParser
+    {
+        public static bool TryParse(byte[] utf8Text, int index, out float value, out int bytesConsumed)
+        {
+            // Precondition replacement
+            if (utf8Text.Length < 1 || index < 0 || index >= utf8Text.Length)
+            {
+                value = 0;
+                bytesConsumed = 0;
+                return false;
+            }
+
+            value = 0f;
+            bytesConsumed = 0;
+            string floatString = "";
```

Comment:
Not that it will really matter since this is a temporary method, but we should probably use a StringBuilder here.